monitor/linuxmonitor/processor.go

```diff
@@ -302,9 +255,77 @@ func (s *LinuxProcessor) ReSync(e *rpcmonitor.EventInfo) error {
 // generateContextID creates the contextID from the event information
 func generateContextID(eventInfo *rpcmonitor.EventInfo) (string, error) {
 
-	if eventInfo.PUID == "" {
-		return "", fmt.Errorf("PUID is empty from eventInfo")
+	return eventInfo.PUID, nil
+}
+
+func (s *LinuxProcessor) processLinuxServiceStart(event *rpcmonitor.EventInfo, runtimeInfo *policy.PURuntime) error {
+	list, err := cgnetcls.ListCgroupProcesses(event.PUID)
+	if err == nil {
+		//cgroup exists and pid might be a member
+		isrestart := func() bool {
+			for _, element := range list {
+				if element == event.PID {
+					//pid is already there it is restart
+					return true
+				}
+			}
+			return false
+		}()
+
+		if !isrestart {
+			pid, _ := strconv.Atoi(event.PID)
+			s.netcls.AddProcess(event.PUID, pid) // nolint
+			return nil
+		}
 	}
 
-	return eventInfo.PUID, nil
+	//It is okay to launch this so let us create a cgroup for it
+	err = s.netcls.Creategroup(event.PUID)
+	if err != nil {
+		return err
+	}
+
+	markval := runtimeInfo.Options().CgroupMark
+	if markval == "" {
+		if derr := s.netcls.DeleteCgroup(event.PUID); derr != nil {
+			zap.L().Warn("Failed to clean cgroup", zap.Error(derr))
+		}
+		return errors.New("Mark value not found")
+	}
+
+	mark, _ := strconv.ParseUint(markval, 10, 32)
```

Comment:
Is that some errors you ignore here and below?

Reply:
If the delete files it means that roll back of what we tried to do failed. There is an additional warning there, but the main error is that we didn't do it in the first place. (ie the question is if an error on the roll-back is as important ... we are already in a mess in this case)